load_balancer/load_balancer.js

```diff
@@ -30,56 +33,19 @@ app.get('/userCheck', function (req,res) {
   res.send(true);
 });
 
-app.get('/connectionCheck', function (req,res) {
-  console.log('connectionCheck requested');
-  var result = 'Load Balancer Working\n';
-  var numOfNodes = nodes_data.Nodes.length;
-  function checkNodeConn(node){
-    var options = {
-      host: node.hostname,
-      port: node.port,
-      path: '/connectionCheck',
-      key : fs.readFileSync('./ssl/key.pem'),
-      cert: fs.readFileSync('./ssl/cert.pem'),
-      rejectUnauthorized:false,
-    };
-    //send a get request and capture the response
-    var req = https.request(options, function(res){
-      // Buffer the body entirely for processing as a whole.
-      var bodyChunks = [];
-      res.on('data', function(chunk){
-        bodyChunks.push(chunk);
-      }).on('end', function(){
-        var body = Buffer.concat(bodyChunks);
-        result = result.concat('<br/>Execution Node at '+node.hostname+':'+node.port+' working: ' + body);
-        console.log("nodeing");
-        //return if all requets processed
-        if(--numOfNodes === 0){
-          console.log("DispRes");
-          dispResult();
-        }
-      });
-    });
-    req.on('error', function(e) {
-      result = result.concat('<br/>Execution Node at  '+node.hostname+':'+node.port+' Error: ' + e.message);
-      //return if all requets processed
-      if(--numOfNodes === 0){
-      console.log("DispRes");
-        dispResult();
-      }
-    });
-    req.end();
-  } //checkNodeConnection ends
-
-  function dispResult(){
-    res.send(result);
-  }
-  //Check connection of all nodes
-  for(var i=0;i<nodes_data.Nodes.length;i++)
-  {
-    console.log(numOfNodes);
-    checkNodeConn(nodes_data.Nodes[i]);
-  }
+app.get('/connectionCheck', (req, res) => {
+  console.log('Connection check requested');
+  const lbStatus = nodes_data.load_balancer;
+  lbStatus.status = 'up';
+  status.checkStatus((result) => {
+    //since the request is being processed, it is assumed that load balancer is up
+    const statusJson = result;
```

Comment:
This looks fine, but there is a better library named [lodash](https://lodash.com/) that is good at manipulating json objects. You might want to use it.